multi-node/libvirt/README.md

```diff
@@ -0,0 +1,106 @@
+these to configure the machines and provision TLS artifacts for Kubernetes.
+
+The config drives should contain provisioning scripts and the resources they
+need to operate. Assuming you have cloned `coreos-kubernetes` to $CLONEDIR:
+
+For kube-master:
+```
+cd /var/lib/libvirt/images/coreos
+cp -R $CLONEDIR/multi-node/libvirt/kube-master /var/lib/libvirt/images/coreos/
+cp $CLONEDIR/multi-node/libvirt/certs/kube-apiserver.tar /var/lib/libvirt/images/coreos/kube-master/openstack/latest/
+cp vm_key.pub /var/lib/libvirt/images/coreos/kube-master/openstack/latest/
+
+```
+
+For kube-worker:
+```
+cd /var/lib/libvirt/images/coreos
+cp -R $CLONEDIR/multi-node/libvirt/kube-worker /var/lib/libvirt/images/coreos/
+cp $CLONEDIR/multi-node/libvirt/certs/kube-worker.tar /var/lib/libvirt/images/coreos/kube-worker/openstack/latest/
+cp vm_key.pub /var/lib/libvirt/images/coreos/kube-worker/openstack/latest/
+```
+
+The `user_data` scripts will install the necessary Kubernetes manifests and
+assume a certain preconfigured network structure.
+
+## SELinux configuration
+
+If you are using SELinux in enforcing mode, you will need to properly label
+these config drive directories.
+
+```
+cd /var/lib/libvirt/images/coreos
+chcon -R -t svirt_image_t kube-master
+chcon -R -t svirt_image_t kube-worker
+```
+
+## Network configuration
+
+```
+# virsh net-define $CLONEDIR/multi-node/libvirt/network-kubernetes.xml
+# virsh net-autostart kubernetes
+# virsh net-start kubernetes
+```
+
+## Virtual machine startup
+
+kube-master
+```
+virt-install --connect qemu:///system --import --name kube-master --ram 1024 --vcpus 1 --os-type=linux --os-variant=virtio26 --disk path=/var/lib/libvirt/images/coreos/kube-master.qcow2,format=qcow2,bus=virtio --filesystem /var/lib/libvirt/images/coreos/kube-master/,config-2,type=mount,mode=squash --network bridge=virbrk8s,mac=54:52:00:fe:b3:b0 --vnc --noautoconsole
+```
+
+kube-worker
+```
+virt-install --connect qemu:///system --import --name kube-worker --ram 1024 --vcpus 1 --os-type=linux --os-variant=virtio26 --disk path=/var/lib/libvirt/images/coreos/kube-worker.qcow2,format=qcow2,bus=virtio --filesystem /var/lib/libvirt/images/coreos/kube-worker/,config-2,type=mount,mode=squash --network bridge=virbrk8s,mac=54:52:00:fe:b3:b1 --vnc --noautoconsole
+```
+
+## Configure kubectl
+Assuming you have cloned `coreos-kubernetes` to $CLONEDIR and run `gentls.sh`:
+```
+MASTER_HOST=192.168.120.10 CA_CERT=$CLONEDIR/multi-node/libvirt/certs/ca.pem ADMIN_KEY=$CLONEDIR/multi-node/libvirt/certs/admin-key.pem ADMIN_CERT=$CLONEDIR/multi-node/libvirt/certs/admin.pem
+kubectl config set-cluster default-cluster --server=https://${MASTER_HOST} --certificate-authority=${CA_CERT}
+kubectl config set-credentials default-admin --certificate-authority=${CA_CERT} --client-key=${ADMIN_KEY} --client-certificate=${ADMIN_CERT}
+kubectl config set-context default-system --cluster=default-cluster --user=default-admin
+kubectl config use-context default-system
```

Comment:
nit: A user may have many other contexts already in their kubeconfig. Can we use something a little more descriptive? coreos-libvirt maybe?